cli/azd/internal/vsrpc/server.go

```diff
@@ -54,6 +61,19 @@ func (s *Server) Serve(l net.Listener) error {
 		mux.Handle("/TestDebugService/v1.0", newDebugService())
 	}
 
+	// Run upload periodically in the background while the server is running.
+	ts := telemetry.GetTelemetrySystem()
+	go func() {
+		ticker := time.NewTicker(5 * time.Second)
+		for {
+			err := ts.RunBackgroundUpload(context.Background(), false)
+			if err != nil {
+				log.Printf("telemetry upload failed: %v", err)
+			}
+			<-ticker.C
+		}
```

Comment:
Do we need something to stop this func from running forever? Perhaps leveraging go context with cancellation that can be checked for? I'm assuming otherwise it just runs until the process is terminated?

Reply:
That's a great question.  The state that is being processed by the upload func is largely persisted in the filesystem and checkpointed enough by design. It is perfectly fine to perform a non-graceful shutdown and things would resume as "normal".